### DISCUSSION TITLE
Don’t notify for transient network errors

### DISCUSSION THREAD

**Discussion by SimonSapin** [bitfireAT/icsx5#60] [Ideas]
Related to https://github.com/bitfireAT/icsx5/discussions/45 but not quite the same.

I semi-regularly get notifications from ICSx⁵ (version 2.0.2-gplay) that say:

> Synchronization error
>
> Unable to resolve host "calendar.google.com": No address associated with hostname

If I then go into the app and trigger another synchronization, a spinner is shown for a while and then the error notification goes away. I suspect the error happens when mobile reception is bad and the app temporarily cannot access the internet. It seems to happen only when I’m not at home (where I have reliable wifi). I can reproduce the same error message with a manual trigger while in airplane mode.

It’d be nicer not to show a notification for such cases when the server is (probably) temporarily unreachable, as opposed to having gone away entirely. For example:

* Don’t auto-update while the Android connectivity API says the device does not have internet access
* If a calendar has been successfully synchronized before, on network-related errors, silently try again a few minutes/hours later and only notify if errors keep happening

**Comment by yan12125** [comment]
> If a calendar has been successfully synchronized before, on network-related errors, silently try again a few minutes/hours later and only notify if errors keep happening

+1 for this. I'd like to also ignore transient errors for temporarily unavailable server. For example, one of my subscribed calendar is on a server (which is out of my control) that reports HTTP 503 from time to time.

**Comment by ArnyminerZ** [comment]
This error is detected, and has been discussed on #72 and kind of been solved on #83. We are just waiting if there's a more elegant solution, or a fix by Google.

---

@SimonSapin:
> Don’t auto-update while the Android connectivity API says the device does not have internet access

The issue is that the information provided by Android is wrong. We are already telling the worker to run only when Internet is available:
https://github.com/bitfireAT/icsx5/blob/a3e8c31f2170afd38a31e3d8cb65a8477ade46ab/app/src/main/java/at/bitfire/icsdroid/SyncWorker.kt#L56
But there's some kind of bug in Android which provides a wrong state, or a network change occurs before having time to fetch the new data from the server.

I agree that maybe the solution might be not showing the error notification all the time, but I think, and I believe that @rfc2822 thinks the same, the best option would be to really find a fix, and not have workarounds like this. However, it's something that can surely be discussed.

---

@yan12125:
> I'd like to also ignore transient errors for temporarily unavailable server

Such an specific use case, but maybe we can think about this... It surely can be done.

**Reply by yan12125** [reply]
Thanks! I will look into #83.

**Reply by yan12125** [reply]
Although #83 is closed, I'm still interested in it, as that change achieves better user experience not only for problematic clients, but also problematic servers. May I continue the work in a new pull request?

**Reply by rfc2822** [reply]
Retries and back-off would be interesting; but is there a use for initial delay?

**Reply by yan12125** [reply]
An initial delay is useful for problematic clients but may not be useful for problematic servers. I can try adding a delay only after a failed request.
